pkg/trace/agent/agent.go

```diff
@@ -273,7 +275,10 @@ func (a *Agent) loop() {
 		a.obfuscator,
 		a.DebugServer,
 	} {
-		stopper.Stop()
+		// Fun with golang nil checks
+		if stopper != nil && !reflect.ValueOf(stopper).IsNil() {
```

Comment:
part of me thinks this reflection check isn't the _ideal_ place to check this (I do recall the internal chat about this) - since technically there could exist a component that performs valid shutdown logic even on a nil pointer. I'm not super strong feeling here though, so this change isn't necessary for merging here.